### PR TITLE
Fix for #39 and #41, and a few small code-style improvements.  

### PR DESCRIPTION
**Pull request by bbaassssiiee**
RedHat support most important.

### CODE REVIEW

**Comment by mariussturm** [comment]
Hi,
thanks for the great work!
Before we can merge this the elastic.ansible-elasticsearch role needs to be on Galaxy to allow easy installation: https://github.com/elastic/ansible-elasticsearch/issues/89
When I apply the role to an Ubuntu host I get a Java 7 installation what is not supported by Graylog, could please fix this before?
Another problem are the tabs in the Readme, when I cut&paste this: https://github.com/Graylog2/graylog-ansible-role/pull/42/commits/96ad0534f7f15c32cfcc822b51c2ac0ee2fc54f5#diff-04c6e90faac2675aa89e2176d2eec7d8R31
Ansible complains because of a yaml error. Could you please use double-white-space indention?

**Comment by bbaassssiiee** [comment]
Things I can fix are done. Now it is up to @elastic.

File: tasks/prepare.yml [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
-#  apt_key: keyserver={{ graylog_java_repo_keyserver }} id={{ graylog_java_repo_key }}
-#  when: ansible_distribution == "Debian"
+- name: Accept apt key for java repo
+  apt_key: keyserver={{ graylog_java_repo_keyserver }} id={{ graylog_java_repo_key }}
+  when: ansible_distribution == "Debian"
 
-#- name: Accept Oracle license prior JDK installation
-#  shell: echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections; echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections creates=/usr/lib/jvm/java-8-oracle
+- name: Accept Oracle license prior JDK installation
+  shell: echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections; echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections creates=/usr/lib/jvm/java-8-oracle
```
Comment (mariussturm) [inline tasks/prepare.yml]: when: ansible_os_family == 'Debian'

File: tasks/prepare.yml [hunk 2]
```diff
@@ -18,15 +18,15 @@
...
 
-#- name: Install Java 8
-#  apt: name=oracle-java8-installer state=present force=yes
+- name: Install Java 8
+  apt: name=oracle-java8-installer state=present force=yes
```
Comment (mariussturm) [inline tasks/prepare.yml]: when: ansible_os_family == 'Debian'

**Comment by mariussturm** [comment]
Just double checked the installation on Centos7. Two things were failing

  * MongoDb was not started automatically
  * The reverted Java installation needs to be conditional with `when: ansible_os_family == 'Debian'`

**Comment by mariussturm** [comment]
Personally, I would prefer to split the Ubuntu installation and the RedHat installation in separate files and load them with conditional imports: http://docs.ansible.com/ansible/playbooks_conditionals.html#conditional-imports
In that way you dont need a when clause on every resource.

**Comment by bbaassssiiee** [comment]
Java is conditional, and the OS specific actions are split in 2 files.
I made a PR to add handlers in lesmyrmidons.mongodb, then we can call mongo to start.

File: defaults/main.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ graylog_java_repo_keyserver: 'keyserver.ubuntu.com'
 graylog_java_repo_key: 'EEA14886'
 
 graylog_packages_url: 'http://packages.graylog2.org/repo/debian/pool/stable/2.1/g/graylog-2.1-repository/graylog-2.1-repository_1-1_all.deb'
-
+graylog_yum_repo_url: 'https://packages.graylog2.org/repo/packages/graylog-2.1-repository_latest.rpm'
```
Comment (danvaida) [inline defaults/main.yml]: this var name is misleading. I'd suggest something like `graylog_yum_rpm_url`.
furthermore, I'd compose it using two vars. for example:
```
graylog_version: '2.1'
graylog_yum_rpm_url: "https://packages.graylog2.org/repo/packages/graylog-{{ graylog_version }}-repository_latest.rpm"
```
you get the idea...

**Comment by danvaida** [comment]
@bbaassssiiee I'd like to propose an amendment to your PR. It's more tied to standards and consistency, in general: replace occurrences of 'yes' with 'True', false with 'False' (notice case and single quotation).
Cheers,
Dan.

**Comment by bbaassssiiee** [comment]
True and False are Booleans, you should not quote them (makes them strings).

**Comment by danvaida** [comment]
hmm... if you suspect that I didn't know that, let me be abundantly clear then:
yes becomes True
Yes becomes True
false becomes False
False becomes False

**Comment by mariussturm** [comment]
@danvaida @bbaassssiiee this is fine but please keep the focus on the centos support and not refactor additional parts of the role. It's better to create a second PR for things like this. When the Elasticsearch role is ready I want to merge this.

**Comment by bbaassssiiee** [comment]
I would like to leave it to Graylog employees, I think I added enough.

**Comment by mariussturm** [comment]
@bbaassssiiee yes, thanks for your great work!

**Comment by bbaassssiiee** [comment]
Actually you don't need to wait for Elastic when they add their role to Galaxy because this PR pulls from GitHub. If you merge you get RedHat support.

**Comment by mariussturm** [comment]
Couldn't make it work with the documented `ansible-galaxy install -p . Graylog2.graylog-ansible-role` command? Did you try that?

**Comment by danvaida** [comment]
is this PR going to be merged anytime soon? I'm waiting for it to update my fork's master and then make a PR myself...

**Comment by mariussturm** [comment]
@danvaida still waiting for the ES guys, maybe you can bump the ticket.

**Comment by bbaassssiiee** [comment]
lesmyrmidons.mongodb merged my PR which added handlers to restart.

**Comment by mariussturm** [comment]
@bbaassssiiee could you please rebase your branch. I just merge the PR and we go with the Github download for the ES role. I am bored of waiting for the Elastic guys.

**Comment by mariussturm** [comment]
Thanks a lot for your great work and your patencie!! 🎉

**Comment by mohanrao** [comment]
So this means latest the role on the galaxy has this PR and it works for redhat?

**Comment by bbaassssiiee** [comment]
I added support for redhat, and this runs from the master branch.

**Pull request merged**